cpu/stm32_common/cpu_init.c

```diff
@@ -49,6 +50,44 @@ void cpu_init(void)
     periph_clk_en(APB1, BIT_APB_PWREN);
     /* initialize the system clock as configured in the periph_conf.h */
     stmclk_init_sysclk();
+
+#if defined(CPU_FAM_STM32L1)
+    uint32_t ahb_gpio_clocks;
+    GPIO_TypeDef *port;
+
+    /* enable GPIO clock and save GPIO clock configuration */
+    ahb_gpio_clocks = RCC->AHBENR & 0xFF;
+    periph_clk_en(AHB, 0xFF);
+
+    /* switch all GPIOs to AIN mode to minimize power consumption */
+    /* can't be more than 12 ports on STM32L1 */
+    for (int i = 0; i < 12; i++) {
```

Comment:
Why 12 ? It seems to me that there are 8 ports on stm32l1. Do you have any reference for this ?